### DISCUSSION TITLE
How to map a domain name to AKS load balancer external IP

### DISCUSSION THREAD

**Discussion by hasithamgh** [Azure/AKS#2382] [Q&A]
I have deployed few APIs to ASK by using Azure DevOps pipelines. the repositories are deployed correctly and I can access the services by the external IP. I need to map these IPs to a domain name which I purchased. 

I tried both of the following links but didn't succeed. 

1. [Use a static public IP address and DNS label with the Azure Kubernetes Service (AKS) load balancer](https://docs.microsoft.com/en-us/azure/aks/static-ip)
2. [Create an ingress controller with a static public IP address in Azure Kubernetes Service (AKS)](https://docs.microsoft.com/en-us/azure/aks/ingress-static-ip)

Highly appreciate if someone could explain the steps of mapping a domain name to AKS external IPs.
Thanks!

**Comment by ramirezalfredo** [comment]
Hello @hasithamgh, if you're a domain owner, you can host it on [Azure DNS](https://docs.microsoft.com/en-us/azure/dns/dns-overview) and use [External DNS](https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/azure.md) in AKS. I'm assuming that you're using an Ingress Controller as well. I have followed that tutorial storing the service principal in a Secret and works very well with Azure DNS.

**Reply by PixelRobots** [reply]
Hopefully, this guide will help. https://pixelrobots.co.uk/2021/06/externaldns-and-azure-kubernetes-service-aks/
